src/database/queries/tag.js

```diff
@@ -0,0 +1,21 @@
+import knex from '../knex'
+import * as _ from './utilities'
+
+const add = attributes =>
+  _.createRecord( 'tag', attributes )
+
+const deleteAll = () =>
+  _.deleteAll( 'tag' )
+
+const getAll = () =>
+  _.findAll( 'tag' )
+  .orderBy('names', 'asc')
+
+const getBy = ( column, data ) =>
+  _.findAllWhere( 'tag', column, data )
+
+const expunge = ( column, data ) =>
+  _.deleteRecord( 'tag', column, data )
+
+
+export { add, deleteAll, getAll, getBy, expunge }
```

Comment:
A comment on the API you've build here: If looks like your tags are just single unique strings. Why not just expose those strings and hide all other SQL stuffs. 

```js
export const addTag(tagName) => 
  _.createRecord('tags', {name: tagName})
    .then(record => record.name)

export const removeTag(tagName) =>
  _.deleteRecord('tags', 'name', tagName)

export const getAllTags() =>
  _.findAll('tags')
    .orderBy('name', 'asc')
    .map(tag => tag.name)

```

With this API nothing but the tag name itself is ever used or exposed. The SQL database ID is completely hidden behind this simpler api. Which I think can be used to do all the things you're doing in your tests.